Branch-SDK/src/main/java/io/branch/referral/BranchActivityLifecycleObserver.java

```diff
@@ -66,6 +73,10 @@ public void onActivityResumed(@NonNull Activity activity) {
             // and the entry Activity is not the launcher Activity where user placed initSession themselves.
             Branch.sessionBuilder(activity).init();
         }
+
+        // must be called after session initialization, which relies on checking whether activity
+        // that is initializing the session is being launched from stack or anew
+        activitiesOnStack.add(activity.getLocalClassName());
```

Comment:
Activities are added, but never removed.   Further, It's not obvious from the ticket why this is being addressed in this PR.

Reply:
This is part of SDK-672, fixing the instant deeplinking, I linked the two issues on Jira. This was a good catch though because, when app is in background and being reopened via a deeplink or home screen icon, Activity does get recreated apparently, so I added removal of activities to onCreate.